### PR TITLE
[Berkeley] Remove special-casing of lookup-related columns

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds upon https://github.com/MinaProtocol/mina/pull/13546, with proof-systems counterpart https://github.com/o1-labs/proof-systems/pull/1117. This removes the special-casing around the lookup argument, allowing us to fix the issues with the feature flags for the various parts of the argument.

As part of these fixes, the lookup sorted columns are now handled correctly for the different widths used by the various different gates.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml [hunk 1]
```diff
@@ -270,7 +268,12 @@ module Make (Inputs : Inputs_intf) = struct
     ; foreign_field_mul_selector
     ; xor_selector
     ; rot_selector
-    ; lookup = Option.map ~f:lookup_eval_of_backend lookup
+    ; lookup_aggregation
+    ; lookup_table
+    ; lookup_sorted =
+        Array.init 5 ~f:(fun i ->
```
Comment (rbonichon) [inline src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml:274]: `5` seems to be a little bit magic here. Could there be an explanation in the form of comments?

File: src/lib/pickles/evaluation_lengths.ml [hunk 2]
```diff
@@ -22,5 +22,8 @@ let create ~of_int =
     ; foreign_field_mul_selector = None
     ; xor_selector = None
     ; rot_selector = None
-    ; lookup = None
+    ; lookup_aggregation = None
+    ; lookup_table = None
+    ; lookup_sorted = Array.init 5 ~f:(fun _ -> None)
```
Comment (rbonichon) [inline src/lib/pickles/evaluation_lengths.ml:27]: Ditto

File: src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml [hunk 3]
```diff
@@ -347,7 +348,12 @@ module Make (Inputs : Inputs_intf) = struct
     ; foreign_field_mul_selector
     ; xor_selector
     ; rot_selector
-    ; lookup = Option.map ~f:lookup_eval_to_backend lookup
+    ; lookup_aggregation
+    ; lookup_table
+    ; lookup_sorted =
+        Array.init 5 ~f:(fun i ->
```
Comment (rbonichon) [inline src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml:354]: Ditto

**Review by rbonichon — APPROVED** [review]
LGTM. Would be nice to document occurences of `5`

File: src/lib/pickles_types/plonk_types.ml [hunk 4]
```diff
@@ -484,16 +419,14 @@ module Evals = struct
         ; foreign_field_mul_selector
         ; xor_selector
         ; rot_selector
+        ; lookup_aggregation
+        ; lookup_table
+        ; runtime_lookup_table
```
Comment (fabrizio-m) [inline src/lib/pickles_types/plonk_types.ml:424]: I'm missing some of the context, but is it ok for `runtime_lookup_table` to be twice?

**Review by fabrizio-m — APPROVED** [review]

File: src/lib/pickles_types/plonk_types.ml (hunk 4 above)
Comment (mrmr1993) [reply src/lib/pickles_types/plonk_types.ml:424]: This looks like an error, yes.

**Review by querolita — APPROVED** [review]

File: src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml (hunk 1 above)
Comment (mrmr1993) [reply src/lib/crypto/kimchi_backend/common/plonk_dlog_proof.ml:274]: Sure. Max number of columns in the sorted lookup argument

File: src/lib/pickles_types/plonk_types.ml (hunk 4 above)
Comment (rbonichon) [reply src/lib/pickles_types/plonk_types.ml]: Fixed

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**